tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```

Comment:
the target should be `esnext`, no?

Reply:
This is related to the "build output" of type script. The question is if we need this in the _minimal_ setup for testing anyways? I am going to test it.

Reply:
I've created the typescript minimal configuration with `tsc --init`. The _minimal and default_ setup of typescript will be used in this PR.